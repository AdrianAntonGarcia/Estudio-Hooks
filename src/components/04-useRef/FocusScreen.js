import React, { useRef } from "react";
import "../02-useEffect/effects.css";

export const FocusScreen = () => {
	// Hacemos referencias a cualquier elemento html sin recargar el componente
	const inputRef = useRef();
	// console.log(ref);

	const handleClick = () => {
		// document.querySelector("input").select();
		inputRef.current.select();
		console.log(inputRef);
	};
	return (
		<div>
			<h1>Focus Screen</h1>
			<hr />
			<input ref={inputRef} className="form-control" placeholder="Su nombre" />
			<button
				onClick={handleClick}
				className="btn btn-outline-primary mt-5"
			>
				Focus
			</button>
		</div>
	);
};
