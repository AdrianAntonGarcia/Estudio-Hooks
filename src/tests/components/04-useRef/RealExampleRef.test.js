import React from "react";
import { RealExampleRef } from "../../../components/04-useRef/RealExampleRef";
import { shallow } from "enzyme";

describe("Pruebas en RealExampleRef", () => {
	test("Debe mostrarse correctamente", () => {
		const wrapper = shallow(<RealExampleRef />);
		expect(wrapper).toMatchSnapshot();
	});

	test("Debe de mostrar el componente MultipleCustomHooks", () => {
		const wrapper = shallow(<RealExampleRef />);
		expect(wrapper.find("button").simulate("click"));
		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find("MultipleCustomHooks").exists()).toBe(true);
	});
});
