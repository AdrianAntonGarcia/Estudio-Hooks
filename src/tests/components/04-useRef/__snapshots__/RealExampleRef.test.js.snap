// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en RealExampleRef Debe de mostrar el componente MultipleCustomHooks 1`] = `
<div>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <MultipleCustomHooks />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show/hide
  </button>
</div>
`;

exports[`Pruebas en RealExampleRef Debe mostrarse correctamente 1`] = `
<div>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show/hide
  </button>
</div>
`;
